commands/cloudapi-v6/volume_test.go

```diff
@@ -1028,194 +1030,253 @@ func TestRunServerVolumeGetErr(t *testing.T) {
-		viper.Set(core.GetFlagName(cfg.NS, cloudapiv6.ArgServerId), testServerVar)
-		viper.Set(core.GetFlagName(cfg.NS, cloudapiv6.ArgAll), true)
-		viper.Set(core.GetFlagName(cfg.NS, config.ArgWaitForRequest), false)
-		rm.CloudApiV6Mocks.Server.EXPECT().ListVolumes(testServerVar, testServerVar, resources.ListQueryParams{}).Return(vsAttachedList, nil, nil)
-		rm.CloudApiV6Mocks.Server.EXPECT().DetachVolume(testServerVar, testServerVar, testServerVar, resources.QueryParams{}).Return(&testResponse, testVolumeErr)
-		rm.CloudApiV6Mocks.Server.EXPECT().DetachVolume(testServerVar, testServerVar, testServerVar, resources.QueryParams{}).Return(&testResponse, nil)
-		err := RunServerVolumeDetach(cfg)
-		assert.Error(t, err)
-	})
-}
+			// Expected gomock calls, call order, call counts and returned values
+			tc.calls()
 
-func TestRunServerVolumeDetachErr(t *testing.T) {
-	var b bytes.Buffer
-	w := bufio.NewWriter(&b)
-	core.CmdConfigTest(t, w, func(cfg *core.CommandConfig, rm *core.ResourcesMocksTest) {
-		viper.Reset()
-		viper.Set(config.ArgOutput, config.DefaultOutputFormat)
-		viper.Set(config.ArgQuiet, false)
-		viper.Set(config.ArgForce, true)
-		viper.Set(core.GetFlagName(cfg.NS, cloudapiv6.ArgDataCenterId), testServerVar)
-		viper.Set(core.GetFlagName(cfg.NS, cloudapiv6.ArgServerId), testServerVar)
-		viper.Set(core.GetFlagName(cfg.NS, cloudapiv6.ArgVolumeId), testServerVar)
-		viper.Set(core.GetFlagName(cfg.NS, config.ArgWaitForRequest), false)
-		rm.CloudApiV6Mocks.Server.EXPECT().DetachVolume(testServerVar, testServerVar, testServerVar, resources.QueryParams{}).Return(&testResponse, testVolumeErr)
-		err := RunServerVolumeDetach(cfg)
-		assert.Error(t, err)
-	})
-}
+			err := funcToTest(cfg)
 
-func TestRunServerVolumeDetachWaitErr(t *testing.T) {
-	var b bytes.Buffer
-	w := bufio.NewWriter(&b)
-	core.CmdConfigTest(t, w, func(cfg *core.CommandConfig, rm *core.ResourcesMocksTest) {
-		viper.Reset()
-		viper.Set(config.ArgOutput, config.DefaultOutputFormat)
-		viper.Set(config.ArgQuiet, false)
-		viper.Set(config.ArgForce, true)
-		viper.Set(core.GetFlagName(cfg.NS, cloudapiv6.ArgDataCenterId), testServerVar)
-		viper.Set(core.GetFlagName(cfg.NS, cloudapiv6.ArgServerId), testServerVar)
-		viper.Set(core.GetFlagName(cfg.NS, cloudapiv6.ArgVolumeId), testServerVar)
-		viper.Set(core.GetFlagName(cfg.NS, config.ArgWaitForRequest), true)
-		rm.CloudApiV6Mocks.Server.EXPECT().DetachVolume(testServerVar, testServerVar, testServerVar, resources.QueryParams{}).Return(&testResponse, nil)
-		rm.CloudApiV6Mocks.Request.EXPECT().GetStatus(testRequestIdVar, resources.QueryParams{}).Return(&testRequestStatus, nil, testRequestErr)
-		err := RunServerVolumeDetach(cfg)
-		assert.Error(t, err)
-	})
+			if tc.err {
+				assert.Error(t, err, fmt.Sprintf("Test case '%s' expected an error but got none", tc.name))
+			} else {
+				assert.NoError(t, err, fmt.Sprintf("Test case '%s' expected no error but got %s", tc.name, err))
```

Comment:
I think assert.NoError already says something like - `Received unexpected error:\n%+v", err` by default? Same for assert.Error